src/cinder/Stream.cpp

```diff
@@ -792,17 +792,105 @@ StreamExc::StreamExc( const std::string &fontName ) throw()
 
 /////////////////////////////////////////////////////////////////////
 
-#define STREAM_PROTOTYPES(r,data,T)\
```

Comment:
Maybe we can leverage STREAM_PROTOTYPES() to reduce code duplication?

```
STREAM_PROTOTYPES(int8_t)
STREAM_PROTOTYPES(uint8_t)
STREAM_PROTOTYPES(blabla_t)
```

Reply:
Kind of embarrassed to admit that didn't even occur to me, mostly because i avoid macros like the plague, but also these were generated by a script I wrote that pulled the definitions out of the preprocessed source, but if the preference is to tuck them back behind macros i can submit a different pull request.

Reply:
My concern is if one day new interfaces were added, you would have to copy-paste at multiple lines.